server.py

```diff
@@ -57,14 +59,17 @@ async def create_url(request: Request):
     except ValueError:
         logging.exception(f"returning 422 due to invalid alias of \"{alias}\"")
         raise HTTPException(status_code=HttpResponse.INVALID_ARGUMENT_EXCEPTION.code)
-   
+
 @app.get("/list")
-async def get_all_urls():
+async def get_all_urls(search: Optional[str] = None, page: Optional[int] = 1):
+    if page is None or page < 1:
```

Comment:
lets do pages indexed at 0, so in the url, `?page=0` means the first page

Reply:
done